src/AccessibilityInsights.SetupLibrary/VersionSwitcherWrapper.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+
+namespace AccessibilityInsights.SetupLibrary
+{
+    /// <summary>
+    /// Methods to provide support when switching versions (upgrading or changing channels)
+    /// </summary>
+    public static class VersionSwitcherWrapper
+    {
+        /// <summary>
+        /// Installs a more recent version in response to an upgrade (retain the same channel)
+        /// </summary>
+        /// <param name="installerUri">The uri to the web-hosted installer</param>
+        /// <returns>true if the Version Switcher process started successfully, false if not</returns>
+        public static bool InstallUpgrade(Uri installerUri)
+        {
+            return DownloadAndInstall(installerUri, null);
+        }
+
+        /// <summary>
+        /// Installs a different version in response to a channel change
+        /// </summary>
+        /// <param name="installerUri">The uri to the web-hosted installer</param>
+        /// <param name="newChannel">The new channel to use</param>
+        /// <returns>true if the Version Switcher process started successfully, false if not</returns>
+        public static bool ChangeChannel(Uri installerUri, string newChannel)
+        {
+            return DownloadAndInstall(installerUri, newChannel);
+        }
+
+        /// <summary>
+        /// Private method that does the work shared between InstallUpgrade and ChangeChannel
+        /// </summary>
+        /// <param name="installerUrl">The uri to the web-hosted installer</param>
+        /// <param name="newChannel">If not null, the new channel to select</param>
+        /// <returns>true if the Version Switcher process started successfully, false if not</returns>
+        private static bool DownloadAndInstall(Uri installerUri, string newChannel)
+        {
+            List<FileStream> fileLocks = new List<FileStream>();
+            try
+            {
+                string temporaryFolder = GetTemporaryVersionSwitcherFolder();
+                string installedFolder = GetInstalledVersionSwitcherFolder();
+                RemoveFolder(temporaryFolder);
```

Comment:
small comment - upon first read I thought we were creating the temp folder & immediately deleting it; not sure if any renames or rearranging would help